src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -656,6 +644,18 @@ private ParenthesizedExpressionSyntax GetParenthesizedExpressionSyntax(Token ope
             return new ParenthesizedExpressionSyntax(openParen, bodyExpression, closeParen);
         }
 
+        private VariableBlockSyntax GetVariableBlock(Token openParen, ImmutableArray<SyntaxBase> expressionsOrCommas, SyntaxBase closeParen)
```

Comment:
>GetVariableBlock [](http://example.com/codeflow?start=36&length=16)

Do we need to update the pseudo grammar?

Reply:
As discussed, I'm going to keep the existing behavior of only allowing 2 args in a `for` expression. So no need to update the grammar